test-integ/peering_commontopo/ac5_2_test.go

```diff
@@ -0,0 +1,315 @@
+	var partition = "default"
+	dc := ct.Sprawl.Topology().Clusters[s.DC]
+	apiClient := ct.APIClientForCluster(t, dc)
+
+	dc3 := ct.Sprawl.Topology().Clusters[ct.DC3.Name]
+	s.dc3APIClient = ct.APIClientForCluster(t, dc3)
+
+	// get peer names for dc2 & dc3 cluster respectively
+	var peers []string
+	peers = append(peers, LocalPeerName(ct.DC2, partition))
+	peers = append(peers, LocalPeerName(ct.DC3, partition))
+
+	s.testServiceHealthCheck(t, apiClient)
+	s.testQueryTwoFailovers(t, apiClient, ct, peers)
+}
+
+func (s *preparedQueryFailoverSuite) testServiceHealthCheck(t *testing.T, apiClient *api.Client) {
+	t.Run("validate service health in cluster", func(t *testing.T) {
+		// preconditions check
+		assertServiceHealth(t, s.dc3APIClient, s.serverSID.Name, 1) // validate server is healthy in dc3 cluster
+		assertServiceHealth(t, apiClient, s.serverSID.Name, 1)
+	})
+}
+
+func (s *preparedQueryFailoverSuite) testQueryTwoFailovers(t *testing.T, c *api.Client, ct *commonTopo, peers []string) {
+	if s.DC == "dc2" {
+		// TO-DO: currently failing
+		t.Skip()
+	}
+
+	t.Run("prepared query with two failovers", func(t *testing.T) {
+		var err error
+
+		// disable dc1 & dc2 and relaunch topology
+		cfg := ct.Sprawl.Config()
+		require.NoError(t, disableNodeInCluster(t, cfg, ct.DC1.Name, s.serverSID.Name, true)) //dc1
+		require.NoError(t, disableNodeInCluster(t, cfg, ct.DC2.Name, s.serverSID.Name, true)) //dc2
+		require.NoError(t, ct.Sprawl.Relaunch(cfg))
+
+		// assert server health status
+		dc2 := ct.Sprawl.Topology().Clusters[ct.DC2.Name]
+		dc2APIClient := ct.APIClientForCluster(t, dc2)
+
+		assertServiceHealth(t, c, s.serverSID.Name, 0)              //dc1
+		assertServiceHealth(t, dc2APIClient, s.serverSID.Name, 0)   //dc2
+		assertServiceHealth(t, s.dc3APIClient, s.serverSID.Name, 1) //dc3 is status should be unchanged
+
+		// create prepared query definition
+		def = &api.PreparedQueryDefinition{
+			Name: "ac5-prepared-query",
+			Service: api.ServiceQuery{
+				Service:     s.serverSID.Name,
+				Partition:   ConfigEntryPartition(s.serverSID.Partition),
+				OnlyPassing: true,
+				// create failover for peer in dc2 and dc3 cluster
+				Failover: api.QueryFailoverOptions{
+					Targets: func() []api.QueryFailoverTarget {
+						var queryFailoverTargets []api.QueryFailoverTarget
+						for _, peer := range peers {
+							queryFailoverTargets = append(queryFailoverTargets, api.QueryFailoverTarget{
+								Peer: peer,
+							})
+						}
+						return queryFailoverTargets
+					}(),
+				},
+			},
+		}
+
+		query = c.PreparedQuery()
+		def.ID, _, err = query.Create(def, nil)
+		require.NoError(t, err)
+
+		// Read registered query
+		queryDef, _, err := query.Get(def.ID, nil)
+		require.NoError(t, err)
+		require.Len(t, queryDef, 1, "expected exactly 1 prepared query")
+		require.Equal(t, 2, len(queryDef[0].Service.Failover.Targets), "expected 2 failover targets for dc2 & dc3")
+		fmt.Println("PreparedQuery with failover created successfully.")
```

Comment:
I think you could probably cut this out. Unit tests should prove this sort of basic thing, that what we wrote is what we get back on read.